UNIXFS.md

```diff
@@ -59,6 +64,61 @@ For files that are comprised of more than a single block, the 'Type' field will
 
 This data is serialized and placed inside the 'Data' field of the outer merkledag protobuf, which also contains the actual links to the child nodes of this object.
 
+## Metadata
+
+UnixFS currently supports three metadata fields:
+
+* `mimeType` -- The mime-type of the file. This generally shouldn't be used.
+* `mode` -- The `mode` is for optionally persisting the [file permissions in numeric notation](https://en.wikipedia.org/wiki/File_system_permissions#Numeric_notation) \[[spec](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_stat.h.html)\]. It defaults to 0755 if unspecified.
+* `mtime` -- The modification time in seconds since the epoch. This defaults to the unix epoch if unspecified.
+
+There are three ways to specify file metadata:
+
+### Embedded in the directory
+
+Each entry in the repeated metadata field corresponds to the linked file/directory at the same offset in the Links section of the outer dag-pb. This field is appropriate in nodes with type `Directory` and `HAMTShard`. However, metadata should _not_ be specified for links that point to other HAMT shards.
+
+For some context, this solution was chosen over embedding the metadata in the file itself because:
+
+1. It allows accessing the metadata without downloading the target files.
+2. More importantly, it avoids changing the root hash of the target files to allow for better deduplication.
+
+### Using an intermediate node
+
+DEPRECATED
+
+Metadata can be applied to a single file/directory with an intermediate "metadata node":
+
+1. A `Type` of `Metadata`.
+2. A `Data` containing an encoded `Metadata` message.
+3. A single `Link` (in the outer dag-pb node) pointing to the actual file/directory.
+
+This solution has been _deprecated_ because it requires an additional intermediate node just for the metadata. It is also poorly supported.
+
+### Default metadata
+
+The `defaultMetadata` field can be used in conjunction with the repeated `metadata` field to specify metadata "defaults".
+
+1. The first `defaultMetadata` field encountered when traversing a sharded directory takes precedence.
```

Comment:
Somewhat related point, does `defaultMetadata` apply to just the directory it is defined in, or can I specify it in a parent directory and have a parent/sub or parent/sub/sub inherit it? I assume _no_, but maybe worth clarifying?

Reply:
No. But I'd like to revisit https://github.com/ipfs/specs/pull/223#discussion_r337847720 and consider dropping it before moving on.

Reply:
I've dropped it. One byte per file in a directory isn't worth it.